### PR TITLE
[DeadCode] Skip Route annotation on RemoveDelegatingParentCallRector

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/6761

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
Updated to allow attributes as well so it will keep working when annotations migrated to attributes.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveDelegatingParentCallRector/Fixture/skip_required2.php.inc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        parent::setEventDispatcher($eventDispatcher);
+    }
+}
+
+class ParentClassWithoutRequired
+{
+    public function setEventDispatcher(EventDispatcherInterface $eventDispatcher)
+    {
+    }
+}
```
Comment (TomasVotruba) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveDelegatingParentCallRector/Fixture/skip_required2.php.inc]: There should be only 1 class in the fixture. The extra class should be located in `/Source` directory. 
That way we also test the reflection of external class works :+1:

Comment (samsonasik) [reply rules-tests/DeadCode/Rector/ClassMethod/RemoveDelegatingParentCallRector/Fixture/skip_required2.php.inc]: updated https://github.com/rectorphp/rector-src/pull/1138/commits/34437ba5544149301edc055aa75536bd6c49b259

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**